backend/src/apiserver/server/experiment_server.go

```diff
@@ -285,7 +285,7 @@ func (s *ExperimentServer) ListExperiments(ctx context.Context, request *apiv2be
 }
 
 func (s *ExperimentServer) deleteExperiment(ctx context.Context, experimentId string) error {
-	err := s.canAccessExperiment(ctx, experimentId, &authorizationv1.ResourceAttributes{Verb: common.RbacResourceVerbArchive})
+	err := s.canAccessExperiment(ctx, experimentId, &authorizationv1.ResourceAttributes{Verb: common.RbacResourceVerbDelete})
```

Comment:
Yes. We can potentially add more fin-grained control to [FakeSubjectAccessReviewClient](https://github.com/gkcalat/pipelines/blob/master/backend/src/apiserver/client/subject_access_review_fake.go#L25-L34). This will require additional changes to all multi-user unit tests.

Another option is to develop end-to-end test for Full Kubeflow with fine-grained RBAC.